application/src/main/java/org/togetherjava/tjbot/commands/filesharing/FileSharingMessageListener.java

```diff
@@ -266,41 +293,18 @@ public UserInteractionType getInteractionType() {
 
     @Override
     public void onButtonClick(ButtonInteractionEvent event, List<String> args) {
-        Member user = event.getMember();
-        String authorId = args.get(0);
+        Member interactionUser = event.getMember();
+        String gistAuthorId = args.get(0);
         boolean hasSoftModPermissions =
-                user.getRoles().stream().map(Role::getName).noneMatch(softModPattern);
+                interactionUser.getRoles().stream().map(Role::getName).anyMatch(isSoftModRole);
 
-        if (!authorId.equals(user.getId()) && hasSoftModPermissions) {
+        if (!gistAuthorId.equals(interactionUser.getId()) && !hasSoftModPermissions) {
             event.reply("You do not have permission for this action.").setEphemeral(true).queue();
             return;
         }
 
         String gistId = args.get(1);
-        HttpRequest request = HttpRequest.newBuilder()
-            .uri(URI.create(SHARE_API + "/" + gistId))
-            .header("Accept", "application/json")
-            .header("Authorization", "token " + gistApiKey)
-            .DELETE()
-            .build();
-
-        HttpResponse<String> apiResponse;
-        try {
-            apiResponse = CLIENT.send(request, HttpResponse.BodyHandlers.ofString());
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
-        } catch (InterruptedException e) {
-            Thread.currentThread().interrupt();
-            throw new IllegalStateException(
-                    "Attempting to delete a gist, but the request got interrupted.", e);
-        }
-
-
-        int status = apiResponse.statusCode();
-        if (status == 404) {
-            throw new IllegalStateException("Gist API unexpected response while deleting gist: %s."
-                .formatted(apiResponse.body()));
-        }
+        deleteGist(gistId);
 
         Message message = event.getMessage();
         List<Button> buttons = message.getButtons();
```

Comment:
I think it might be better to disable the buttons right away, before the long api call.

There is no need for it to wait for it to finish. Advantages:
* u dont need the deferReply thing anymore, since u send ur response right away and do the long stuff afterwards
* the user isnt able to click the buttons multiple times while its doing the api call, since the buttons are disabled right away and not 2 seconds later

Reply:
ok I will do this :D